packages/fxa-auth-server/test/local/routes/linked-accounts.js

```diff
@@ -883,6 +883,78 @@ describe('/linked_account', function () {
       assert.notCalled(mockDB.deleteLinkedAccount);
       assert.notCalled(mockDB.deleteSessionToken);
     });
+
+    it('handles database errors gracefully without unhandled promise rejection', async () => {
+      setupTest({ validateSecurityToken: makeJWT('sessionRevoked') });
+
+      // Mock database to throw an error
+      mockDB.getLinkedAccount = sinon.stub().rejects(new Error('Database connection failed'));
+
+      // This should not throw an unhandled promise rejection
+      await runTest(route, mockRequest);
+
+      assert.calledWithExactly(statsd.increment, 'handleGoogleSET.received');
+      assert.calledWithExactly(statsd.increment, 'handleGoogleSET.decoded');
+      assert.calledWithExactly(
+        statsd.increment,
+        'handleGoogleSET.processing.https://schemas.openid.net/secevent/risc/event-type/sessions-revoked'
+      );
+      // Should not call processed because the event handler failed
+      assert.notCalled(mockDB.sessions);
+      assert.notCalled(mockDB.deleteSessionToken);
+    });
+
+    it('handles session deletion errors gracefully', async () => {
+      setupTest({ validateSecurityToken: makeJWT('sessionRevoked') });
+
+      // Mock database to throw an error during session deletion
+      mockDB.getLinkedAccount = sinon.stub().resolves({ uid: UID });
+      mockDB.sessions = sinon.stub().resolves([
+        { id: 'sessionTokenId1', uid: UID, providerId: 1 },
+        { id: 'sessionTokenId2', uid: UID, providerId: 1 },
+      ]);
+      mockDB.deleteSessionToken = sinon.stub()
+        .onFirstCall().resolves()
+        .onSecondCall().rejects(new Error('Session deletion failed'));
+
+      await runTest(route, mockRequest);
+
+      assert.calledWithExactly(statsd.increment, 'handleGoogleSET.received');
+      assert.calledWithExactly(statsd.increment, 'handleGoogleSET.decoded');
+      assert.calledWithExactly(
+        statsd.increment,
+        'handleGoogleSET.processed.https://schemas.openid.net/secevent/risc/event-type/sessions-revoked'
+      );
+      // Should still process the first session successfully
+      assert.calledWithExactly(mockDB.deleteSessionToken, {
+        id: 'sessionTokenId1',
+        uid: UID,
+        providerId: 1,
+      });
+      assert.calledWithExactly(mockDB.deleteSessionToken, {
+        id: 'sessionTokenId2',
+        uid: UID,
+        providerId: 1,
+      });
+    });
```

Comment:
Do you want add assertions on statsd call that will happen with second call fails?